packages/model-viewer/src/features/annotation.ts

```diff
@@ -124,6 +132,46 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
       this[$needsRender]();
     }
 
+    /**
+     * This method returns in-scene data about a requested hotspot including
+     * its position in screen (canvas) space and its current visibility.
+     */
+    queryHotspot(name: string): HotspotData | null {
+      const hotspot = this[$hotspotMap].get(name);
+      if (hotspot == null) {
+        return null;
+      }
+
+      const position = toVector3D(hotspot.position);
+      const normal = toVector3D(hotspot.normal);
+      const facingCamera = hotspot.facingCamera;
+
+      const scene = this[$scene];
+      const camera = scene.getCamera();
+      const vector = new Vector3();
+
+      vector.setFromMatrixPosition(hotspot.matrixWorld);
+      vector.project(camera);
+
+      const widthHalf = scene.width / 2;
+      const heightHalf = scene.height / 2;
+
+      vector.x = (vector.x * widthHalf) + widthHalf;
+      vector.y = -(vector.y * heightHalf) + heightHalf;
```

Comment:
I wonder if we could pull this info out of the `CSS2DRenderer` instead of duplicating the math here? I can't remember if that all gets hidden by a closure or not.

Reply:
Yeah I wondered this too but it seems the class [does the work and applies the transform css directly, otherwise discarding the result](https://github.com/mrdoob/three.js/blob/dev/examples/jsm/renderers/CSS2DRenderer.js#L112-L154) (Unless I'm missing something)

Reply:
Yeah, extracting it from the CSS sounds worse that computing it. No problem.